org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/impl/BuilderStateDiscarder.java

```diff
@@ -39,6 +43,12 @@ public boolean forgetLastBuildState(Iterable<IProject> toUpdate, Map<String, Str
 				XtextBuilder builder = BuildManagerAccess.findBuilder(project);
 				if (builder != null) {
 					builder.forgetLastBuiltState();
+					try {
+						// Touch the project such that the auto-build knows it should be rebuild
+						project.touch(null);
```

Comment:
maybe have this better patchable in case this leads to problems in some projects

Reply:
How about this?